### PR TITLE
Fix privacy backward compatibility

### PR DESCRIPTION
**Pull request by pan-**
### Description

This PR revert the additions made for the privacy change to BLEProtocol::AddressType in favor of a new type. The function connect as well as the advertising report and connection report have been modified accordingly.

### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

### CODE REVIEW

**Comment by pan-** [comment]
@paul-szczepanek-arm  Could you review this PR ?

File: features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF51/source/nRF5xGap.cpp [hunk 1]
```diff
@@ -259,12 +260,10 @@ ble_error_t nRF5xGap::startAdvertising(const GapAdvertisingParams &params)
     whitelist.addr_count = 0;
     whitelist.irk_count  = 0;
 
-    /* Add missing IRKs to whitelist from the bond table held by the SoftDevice */
-    if (advertisingPolicyMode != Gap::ADV_POLICY_IGNORE_WHITELIST) {
```
Comment (paul-szczepanek-arm) [inline features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF51/source/nRF5xGap.cpp]: has this policy been removed?

Comment (pan-) [reply features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF51/source/nRF5xGap.cpp]: No it hasn't been removed and it is still used line 299.

File: features/FEATURE_BLE/source/generic/GenericGap.cpp [hunk 2]
```diff
@@ -1142,112 +1175,126 @@ void GenericGap::on_advertising_report(const pal::GapAdvertisingReportEvent& e)
...
-            {
+        if(_privacy_enabled &&
+            _central_privacy_configuration.resolution_strategy == CentralPrivacyConfiguration_t::RESOLVE_AND_FILTER &&
+            advertising.address_type == pal::connection_peer_address_type_t::RANDOM_ADDRESS &&
+            is_random_private_resolvable_address(advertising.address.data())
+        ) {
                 // Filter it out
-                continue;
-            }
+            return;
```
Comment (donatieng) [inline features/FEATURE_BLE/source/generic/GenericGap.cpp]: What happens if we have multiple advertising reports?

File: features/FEATURE_BLE/source/generic/GenericGap.cpp [hunk 3]
```diff
@@ -1142,112 +1175,126 @@ void GenericGap::on_advertising_report(const pal::GapAdvertisingReportEvent& e)
...
         }
 
+        // note 1-to-1 conversion between connection_peer_address_type_t and
+        // peer_address_type_t
+        peer_address_type_t peer_address_type =
+            static_cast<peer_address_type_t::type>(advertising.address_type.value());
```
Comment (donatieng) [inline features/FEATURE_BLE/source/generic/GenericGap.cpp]: I guess these types are not likely to deviate?

File: features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF51/source/nRF5xGap.cpp [hunk 4]
```diff
@@ -1248,6 +1248,16 @@ void nRF5xGap::on_connection(Gap::Handle_t handle, const ble_gap_evt_connected_t
 
 #if (NRF_SD_BLE_API_VERSION <= 2)
     bool private_peer_known = evt.irk_match;
+
+    // thanks to softdevice consistencies; addresses are not resolved on the
```
Comment (paul-szczepanek-arm) [inline features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF51/source/nRF5xGap.cpp]: "inconsistencies" but I get your meaning and frustration

File: features/FEATURE_BLE/source/generic/GenericGap.cpp (hunk 2 above)
Comment (pan-) [reply features/FEATURE_BLE/source/generic/GenericGap.cpp]: Good catch; continue is more appropriate here.

File: features/FEATURE_BLE/source/generic/GenericGap.cpp (hunk 3 above)
Comment (donatieng) [reply features/FEATURE_BLE/source/generic/GenericGap.cpp]: 👍

**Comment by paul-szczepanek-arm** [comment]
merging, will fix return->continue as a separate commit on my repo

File: features/FEATURE_BLE/source/generic/GenericGap.cpp (hunk 2 above)
Comment (paul-szczepanek-arm) [reply features/FEATURE_BLE/source/generic/GenericGap.cpp]: fixed in my repo

**Pull request merged**